program.c

```diff
@@ -80,6 +81,90 @@ static int load_erase_tag(xmlNode *node, bool is_nand)
 	return 0;
 }
 
+static struct program *program_load_sparse(struct program *program, int fd)
+{
+	struct program *program_sparse = NULL;
+	struct program *programes_sparse = NULL;
+	struct program *programes_sparse_last = NULL;
+	char tmp[PATH_MAX];
+
+	sparse_header_t sparse_header;
+	unsigned start_sector, chunk_size, chunk_type, chunk_data;
+
+	if (sparse_header_parse(fd, &sparse_header)) {
+		/* ignore false sparse images */
```

Comment:
Isn't this comment misleading? If I read the code correctly we'll be programming the referenced image as a normal (non-sparse) image, not ignoring it.

Reply:
updated to

```
/*
 * If the XML tag "program" contains the attribute 'sparse="true"'
 * for a partition node but lacks a sparse header,
 * it will be validated against the defined partition size.
 * If the sizes match, it is likely that the 'sparse="true"' attribute
 * was set by mistake.
 */
```